cli/flox/src/utils/metrics.rs

```diff
@@ -294,65 +320,55 @@ impl MetricsBuffer {
     ///
     /// This is used when the buffer is pushed to the server
     /// and we start collecting metrics in a new buffer.
-    async fn clear(&mut self) {
-        if let Some(ref file) = self.0 {
-            if let Err(e) = file.set_len(0).await {
-                debug!("Could not truncate metrics buffer file: {e}")
-            }
-        }
-        self.1.clear();
+    fn clear(&mut self) -> Result<()> {
+        self.storage
+            .set_len(0)
+            .context("Could not truncate metrics buffer file")?;
+        self.buffer.clear();
+        Ok(())
     }
 
+    /// Returns an iterator over the entries in the buffer
     fn iter(&self) -> impl Iterator<Item = &MetricEntry> {
-        self.1.iter()
+        self.buffer.iter()
     }
 }
 
-async fn read_metrics_uuid(config: &Config) -> Result<Uuid> {
+fn read_metrics_uuid(config: &Config) -> Result<Uuid> {
     let data_dir = &config.flox.data_dir;
     let uuid_path = data_dir.join(METRICS_UUID_FILE_NAME);
 
-    tokio::fs::File::open(&uuid_path)
-        .or_else(|e| async { Err(e).context("Could not read metrics UUID file") })
-        .and_then(|mut f| async move {
+    File::open(uuid_path)
+        .context("Could not read metrics UUID file")
+        .and_then(|mut f| {
             let mut uuid_str = String::new();
-            f.read_to_string(&mut uuid_str).await?;
+            f.read_to_string(&mut uuid_str)?;
             let uuid_str_trimmed = uuid_str.trim();
             Uuid::try_parse(uuid_str_trimmed).with_context(|| {
                 indoc! {"
                 Could not parse the metrics UUID of this installation in {uuid_path}
             "}
             })
         })
-        .await
 }
 
-async fn add_metric(event: PosthogEvent) -> Result<()> {
+fn add_metric(event: MetricEvent) -> Result<()> {
     let config = Config::parse()?;
 
     if config.flox.disable_metrics {
         return Ok(());
     }
 
-    let uuid = read_metrics_uuid(&config).await?;
-    let mut metrics_buffer = MetricsBuffer::read(&config).await?;
-
-    let now = OffsetDateTime::now_utc();
-
-    let new_entry = MetricEntry::new(event, now);
-    metrics_buffer.push(new_entry).await?;
+    let uuid = read_metrics_uuid(&config)?;
+    let mut metrics_buffer = MetricsBuffer::read(&config.flox.cache_dir)?;
 
-    // Note: assumes the oldest metric entry must come first
-    let buffer_time_passed = metrics_buffer
-        .oldest_timestamp()
-        .map(|oldest| now - oldest > Duration::hours(2))
-        .unwrap_or(false);
+    let new_entry = MetricEntry::new(event, OffsetDateTime::now_utc());
+    metrics_buffer.push(new_entry)?;
 
     let force_flush_buffer = std::env::var("_FLOX_FORCE_FLUSH_METRICS").is_ok();
 
-    if buffer_time_passed || force_flush_buffer {
-        debug!("Pushing buffered metrics");
-        push_metrics(metrics_buffer, uuid).await?;
+    if metrics_buffer.is_expired() || force_flush_buffer {
```

Comment:
nonblocking: do we need to check for network? Or should we at least comment and say an error will bubble up but be ignored?